hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffJob.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.om.snapshot;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import org.apache.hadoop.hdds.utils.db.Codec;
+import org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse.JobStatus;
+
+/**
+ * POJO for Snapshot diff job.
+ */
+public class SnapshotDiffJob {
+  private String jobId;
+  private JobStatus status;
+  private String volume;
+  private String bucket;
+  private String fromSnapshot;
+  private String toSnapshot;
+  private boolean forceFullDiff;
+
+  // Default constructor for Jackson Serializer.
+  public SnapshotDiffJob() {
+
+  }
+
+  public SnapshotDiffJob(String jobId,
+                         JobStatus jobStatus,
+                         String volume,
+                         String bucket,
+                         String fromSnapshot,
+                         String toSnapshot,
+                         boolean forceFullDiff) {
+    this.jobId = jobId;
+    this.status = jobStatus;
+    this.volume = volume;
+    this.bucket = bucket;
+    this.fromSnapshot = fromSnapshot;
+    this.toSnapshot = toSnapshot;
+    this.forceFullDiff = forceFullDiff;
```

Comment:
is forceFullDiff coming from the command line ? I guess it is useful to record the type of snapdiff job it is.

Reply:
Yes, it is command line argument.

https://github.com/apache/ozone/blob/1236414c1c46c67c92f37499b0b61921cdfab39a/hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/snapshot/SnapshotDiffHandler.java#L61